modules/ROOT/pages/java-support.adoc

```diff
@@ -231,7 +231,11 @@ To ensure that your applications continue to work as expected, follow these guid
 
 * Verify that the objects used by your application are Plain Old Java Objects (POJOs). 
 +
-POJOs are required for DataWeave 2.6.0 and later, and you must ensure that the POJOs have:
+POJOs are required for DataWeave 2.6.0 and later. Because reflection is more restricted in Java 17, POJOs now require setters. Previously, POJOs had default constructors and getters for all properties. Now, POJOs must also have setters so DataWeave can build outside the connector in the Mule app. 
++
+Constructors and setters are required if your class is instantiated, and getters are required if your class is read. If your class is returned and not instantiated, only getters are required. However, using both getters and setters simplify the validation and certification process.
```

Comment:
Constructors and setters are required if your class is instantiated > Constructors and setters are required if your class is instantiated by DataWeave